src/systems/filecoin_blockchain/storage_power_consensus/storage_power_consensus_subsystem.go

```diff
@@ -86,24 +79,24 @@ func (spc *StoragePowerConsensusSubsystem_I) IsWinningPartialTicket(partialTicke
 	return spc.ec().IsWinningChallengeTicket(challengeTicket, activePowerInSector, totalPower)
 }
 
-func (spc *StoragePowerConsensusSubsystem_I) GetTicketProductionSeed(chain block.Chain, minerAddr addr.Address, epoch block.ChainEpoch) util.Randomness {
-	return chain.RandomnessAtEpoch(minerAddr, epoch-SPC_LOOKBACK_TICKET)
+func (spc *StoragePowerConsensusSubsystem_I) GetTicketProductionRand(chain block.Chain, epoch block.ChainEpoch) util.Randomness {
+	return chain.RandomnessAtEpoch(epoch - SPC_LOOKBACK_TICKET)
 }
 
-func (spc *StoragePowerConsensusSubsystem_I) GetElectionProofSeed(chain block.Chain, minerAddr addr.Address, epoch block.ChainEpoch) util.Randomness {
-	return chain.RandomnessAtEpoch(minerAddr, epoch-SPC_LOOKBACK_RANDOMNESS)
+func (spc *StoragePowerConsensusSubsystem_I) GetElectionProofRand(chain block.Chain, epoch block.ChainEpoch) util.Randomness {
+	return chain.RandomnessAtEpoch(epoch - SPC_LOOKBACK_RANDOMNESS)
 }
 
-func (spc *StoragePowerConsensusSubsystem_I) GetSealSeed(chain block.Chain, minerAddr addr.Address, epoch block.ChainEpoch) util.Randomness {
-	return chain.RandomnessAtEpoch(minerAddr, epoch-SPC_LOOKBACK_SEAL)
+func (spc *StoragePowerConsensusSubsystem_I) GetSealRand(chain block.Chain, epoch block.ChainEpoch) util.Randomness {
+	return chain.RandomnessAtEpoch(epoch - SPC_LOOKBACK_SEAL)
 }
 
-func (spc *StoragePowerConsensusSubsystem_I) GetPoStChallenge(chain block.Chain, minerAddr addr.Address, epoch block.ChainEpoch) util.Randomness {
-	return chain.RandomnessAtEpoch(minerAddr, epoch-SPC_LOOKBACK_POST)
+func (spc *StoragePowerConsensusSubsystem_I) GetPoStChallengeRand(chain block.Chain, epoch block.ChainEpoch) util.Randomness {
+	return chain.RandomnessAtEpoch(epoch - SPC_LOOKBACK_POST)
```

Comment:
These should all use distinct DSTs.

Reply:
Why?

Reply:
(discussion continued below)

Reply:
This is the only one where I strong disagreee,. These methods are just convenience to automatically apply proper offsets for drawing the right ticket. The DSTs should be applied when constructing the seed properly (e.g. PoStRandomness or SealRandomness) as is done (eg) in storage_mining_subsystem for PoSt.

Reply:
What if some of the offsets (`SPC_LOOKBACK_TICKET`, `SPC_LOOKBACK_RANDOMESS`, ...) happen to be equal -- wouldn't this cause us to collide in the randomness output?

Reply:
yes but we apply extra transformations to the randomness (eg for tickets we use epochID, for Posts we use miner addr) so the actual used randomness is processed further. Beside if it wasn't I don't see why that would be so bad, as long as the randomness is deterministically drawn...?

Reply:
I agree that there's no obvious attack -- but I think it would be much less work to prove security if we didn't have to do the case-by-case analysis each time, and instead could just rely on the general composability of the randomness primitive (which only holds if we invoke it with distinct tags for each use).

Reply:
I'm sensitive to that. Let's get feedback from @anorth and @whyrusleeping on #677. If 👍 I'm fine to add. Your point is well taken, I just worry about preemptive complexity and excessive future proofing though that might be an engineering mindset.